openwisp_users/tests/test_admin.py

```diff
@@ -201,6 +201,105 @@ def test_operator_changelist_superuser_column_hidden(self):
+        self._create_org_user(**options2)
+        self.client.force_login(operator)
+        response = self.client.get(reverse('admin:openwisp_users_user_add'))
+        self.assertContains(response, 'operator-org1</option>')
+        self.assertNotContains(response, 'operator-org2</option>')
+
+    def test_operator_change_organization(self):
+        org1 = self._create_org(name='test-org1')
+        org2 = self._create_org(name='test-org2')
+        default_org = Organization.objects.get(name='default')
+        operator = self._create_operator()
+        org_permissions = Permission.objects.filter(codename__endswith='change_organization')
+        operator.user_permissions.add(*org_permissions)
+        options1 = {
+            'organization': org1,
+            'is_admin': True,
+            'user': operator
+        }
+        options2 = {
+            'organization': org2,
+            'is_admin': False,
+            'user': operator
+        }
+        self._create_org_user(**options1)
+        self._create_org_user(**options2)
+        self.client.force_login(operator)
+        response = self.client.get(reverse('admin:openwisp_users_organization_change',
+                                           args=[org1.pk]))
+        self.assertContains(response,
+                            '<input type="text" name="name" value="{0}"'.format(org1.name))
+        response = self.client.get(reverse('admin:openwisp_users_organization_change',
+                                           args=[default_org.pk]))
+        self.assertContains(response,
+                            '<input type="text" name="name" value="{0}"'.format(default_org.name))
+        response = self.client.get(reverse('admin:openwisp_users_organization_change',
+                                           args=[org2.pk]))
+        if response.status_code == 200:
+            self.assertNotContains(response,
+                                   '<input type="text" name="name" value="{0}"'.format(org2.name))
+        else:
+            self.assertEqual(response.status_code, 403)
+
+    def test_operator_create_org_user(self):
+        org1 = self._create_org(name='test-org1')
+        org2 = self._create_org(name='test-org2')
+        operator = self._create_operator()
+        org_permissions = Permission.objects.filter(codename__endswith='change_organization')
+        operator.user_permissions.add(*org_permissions)
+        options1 = {
+            'organization': org1,
+            'is_admin': True,
+            'user': operator
+        }
+        options2 = {
+            'organization': org2,
+            'is_admin': False,
+            'user': operator
+        }
+        self._create_org_user(**options1)
+        self._create_org_user(**options2)
+        self.client.force_login(operator)
+        response = self.client.get(reverse('admin:openwisp_users_organizationuser_add'))
+        self.assertContains(response, '<div class="readonly')
```

Comment:
please make this test more specific, what organization name should be readonly? The name of `org2` right?

Reply:
Here, i was testing if the `is_admin` field is readonly and not the organization `org2`. This was to make sure that operators don't escalate their privileges. I have re-adjust the test and i think now it is more specific. Please if it is not, tell me what is wrong with it so that i can correct it. Thanks